tiledb/sm/array_schema/dimension_label_schema.h

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file tiledb/sm/array_schema/dimension_label_schema.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Defines the array schema class
+ */
+
+#ifndef TILEDB_dimension_label_schema_H
+#define TILEDB_dimension_label_schema_H
+
+#include "tiledb/common/common.h"
+#include "tiledb/sm/enums/label_order.h"
+#include "tiledb/sm/enums/layout.h"
+#include "tiledb/sm/misc/constants.h"
+
+using namespace tiledb::common;
+
+namespace tiledb::type {
+class Range;
+}
+
+using namespace tiledb::type;
+
+namespace tiledb::sm {
+
+class ArraySchema;
+class Attribute;
+class Dimension;
+class ByteVecValue;
+class FilterPipeline;
+
+/**
+ * Return a Status_DimensionLabelSchema error class Status with a given message
+ **/
+inline Status Status_DimensionLabelSchemaError(const std::string& msg) {
+  return {"[TileDB::DimensionLabelSchema] Error", msg};
+}
+
+/**
+ * Schema for an dimension label. An dimension label consists of two
+ * one-dimensional arrays used to define a dimension label.
+ */
+class DimensionLabelSchema {
+ public:
+  /**
+   * Size type for the number of labels in an dimension label and for label
+   * indices.
+   *
+   * This must be the same as ArraySchema::attribute_size_type
+   */
+  using attribute_size_type = unsigned int;
+
+  /** Default constructor is not C.41 compliant. */
+  DimensionLabelSchema() = delete;
+
+  /**
+   * Constructor.
+   *
+   * @param label_type The label type.
+   * @param dim The dimension.
+   * @param label_dim_domain The label domain.
+   * @param label_tile_extent The label tile extent.
+   * @param dim_label The dimension label to be created.
+   * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+   */
+  DimensionLabelSchema(
+      LabelOrder label_order,
+      Dimension* dim,
```

Comment:
I don't think this should be an argument. Instead, we should add only the type and domain of the dimension that this label is created for. That way, it will not depend on the `Dimension` object.

Reply:
Updated.